cam/DeviceSource.swift

```diff
@@ -0,0 +1,38 @@
+import CoreMediaIO
+
+protocol DeviceSourceDelegate: AnyObject {
+    // nothing yet
+}
+
+class DeviceSource: NSObject {
+    let modelName: String
+
+    weak var delegate: DeviceSourceDelegate? = nil
+
+	init(modelName: String) throws {
+        self.modelName = modelName
+	}
+}
+
+extension DeviceSource: CMIOExtensionDeviceSource {
+    var availableProperties: Set<CMIOExtensionProperty> {
+        return [.deviceTransportType, .deviceModel]
+    }
+
+    func deviceProperties(forProperties properties: Set<CMIOExtensionProperty>) throws -> CMIOExtensionDeviceProperties {
+        let deviceProperties = CMIOExtensionDeviceProperties(dictionary: [:])
+
+        if properties.contains(.deviceTransportType) {
+            deviceProperties.transportType = kIOAudioDeviceTransportTypeVirtual
```

Comment:
ooops, getting:

```
Cannot find 'kIOAudioDeviceTransportTypeVirtual' in scope
```

Reply:
Looks like I forgot to pull over the `import IOKit.audio`. Fixed.